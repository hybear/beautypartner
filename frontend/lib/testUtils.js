import casual from 'casual';

// seed it so we get consistent results
casual.seed(777);

const fakeItem = () => ({
  __typename: 'Item',
  id: 'abc123',
  listPrice: 180000,
  bestPrice: 155200,
  category: 100,
  reference: '15523',
  title: 'LA MER The Moisturizing Soft Cream',
  description:
    'An ultra-soft face moisturizer that nurtures for beautiful, glowing radiance, leaving skin looking naturally brightened, hydrated, and restored.',
  image: 'laMer.jpg',
  largeImage: 'largelaMer.jpg',
});

const fakeUser = () => ({
  __typename: 'User',
  id: '4234',
  name: casual.name,
  email: casual.email,
  permissions: ['ADMIN'],
  birthday: '1996-03-08T14:18:00.000Z',
  creditCards: [],
  phone: casual.phone,
  balance: 1800,
  orders: [],
  cart: [],
});

const fakeOrderItem = () => ({
  __typename: 'OrderItem',
  id: casual.uuid,
  image: `${casual.word}.jpg`,
  title: casual.words(),
  listPrice: 4234,
  bestPrice: 4234,
  quantity: 1,
  description: casual.words(),
});

const fakeOrder = () => ({
  __typename: 'Order',
  id: 'ord123',
  charge: 'ch_123',
  total: 400,
  cashback: 5,
  paymentMethod: 'stripe',
  items: [fakeOrderItem(), fakeOrderItem()],
  status: ['APROVED'],
  createdAt: '2020-02-17T01:52:31.269Z',
  user: fakeUser(),
});

const fakeCartItem = overrides => ({
  __typename: 'CartItem',
  id: 'omg123',
  quantity: 3,
  item: fakeItem(),
  user: fakeUser(),
  ...overrides,
});

// Fake LocalStorage
class LocalStorageMock {
  constructor() {
    this.store = {};
  }

  clear() {
    this.store = {};
  }

  getItem(key) {
    return this.store[key] || null;
  }

  setItem(key, value) {
    this.store[key] = value.toString();
  }

  removeItem(key) {
    delete this.store[key];
  }
}

export { LocalStorageMock, fakeItem, fakeUser, fakeCartItem, fakeOrder, fakeOrderItem };
