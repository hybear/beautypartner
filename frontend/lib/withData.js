import withApollo from 'next-with-apollo';
import ApolloClient from 'apollo-boost';
import { endpoint, prodEndpoint } from '../config';

function createClient({ headers }) {
  return new ApolloClient({
    uri: process.env.NODE_ENV === 'development' ? endpoint : prodEndpoint,
    request: operation => {
      operation.setContext({
        fetchOptions: {
          credentials: 'include',
        },
        headers,
      });
    },
    clientState: {
      resolvers: {
        Mutation: {
          // Logout Method to AddToCart
        },
      },
      defaults: {},
    },
  });
}

export default withApollo(createClient);
