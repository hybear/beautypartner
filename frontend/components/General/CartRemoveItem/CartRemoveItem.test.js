import { MockedProvider } from 'react-apollo/test-utils';
import wait from 'waait';
import { ApolloConsumer } from 'react-apollo'
import { act } from 'react-dom/test-utils';
import RemoveFromCart, { REMOVE_FROM_CART_MUTATION } from './index'
import { CURRENT_USER_QUERY } from '../../../User'
import { fakeUser, fakeCartItem } from '../../../../lib/testUtils';

global.alert = console.log;

const mocks = [
    {
        request: { query: CURRENT_USER_QUERY },
        result: { data: { me: {
                ...fakeUser(),
                cart: [fakeCartItem({ id: 'abc123' })],
                }
            }
        }
    },
    {
        request: { query: REMOVE_FROM_CART_MUTATION, variables: { id: 'abc123' }},
        result: { data: {
                removeFromCart: {
                    __typename: 'CartItem',
                    id: 'abc123'
                }
            }
        }
    }
]

describe('<RemoveFromCart/>', () => {
    it('renders and matches snapshots', async () => {
        await act(async () => {
            const wrapper = mount(
                <MockedProvider mocks={mocks}>
                    <RemoveFromCart id="abc123" />
                </MockedProvider>
            )
            expect(wrapper.find('button[data-test="btnRemoveFromCart"]')).toMatchSnapshot();
        });
    });
    it('removes the item from cart', async () => {
        await act(async () => {
            let apolloClient;
            const wrapper = mount(
                <MockedProvider mocks={mocks} addTypename={false}>
                    <ApolloConsumer>
                        {
                            client => {
                                apolloClient = client;
                                return <RemoveFromCart id="abc123" />
                            }
                        }
                    </ApolloConsumer>
                </MockedProvider>
            );
            await wait();
            wrapper.update();

            const res = await apolloClient.query({ query: CURRENT_USER_QUERY })
            expect(res.data.me.cart).toHaveLength(1);
            expect(res.data.me.cart[0].item.bestPrice).toBe(155200);
            wrapper.find('button[data-test="btnRemoveFromCart"]').simulate('click');
            await wait();
            const res2 = await apolloClient.query({ query: CURRENT_USER_QUERY })
            expect(res2.data.me.cart).toHaveLength(0);
        });
    });
});
