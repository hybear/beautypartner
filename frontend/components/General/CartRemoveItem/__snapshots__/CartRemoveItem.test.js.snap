// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RemoveFromCart/> renders and matches snapshots 1`] = `
<button
  className="MuiButtonBase-root MuiButton-root MuiButton-text Button-mxspsd-0 styles__Button-sc-6wf167-0 styles__CartRemoveItemStyles-cmg41v-0 gtFMYh"
  data-test="btnRemoveFromCart"
  disabled={false}
  onBlur={[Function]}
  onClick={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  title="Delete Item"
  type="button"
>
  <span
    className="MuiButton-label"
  >
    Ã—
  </span>
  <NoSsr />
</button>
`;
